### PR TITLE
✨ Add CRUD APIs for user permission

### PR DESCRIPTION
**Pull request by sreehari-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-1862
Issue Type: BUG/ FEATURE

# Changes
- Add `PUT` endpoint to add users to a role/group
- Add `DELETE` endpoint to remove users from role/group

# How has the change been tested?

# Screenshots (if applicable)

# Build Success screenshot (Till a CICD pipeline is set up)

# Notes

### CODE REVIEW

File: forms-flow-api/src/formsflow_api/services/factory/keycloak_group_service.py [hunk 1]
```diff
@@ -144,3 +144,20 @@ def format_response(self, data):
...
+    def add_user_to_group_role(self, user_id: str, group_id: str, payload: Dict):
+        """Add user to group."""
+        data = {
+            "realm": current_app.config.get("KEYCLOAK_URL_REALM"),
+            "userId": payload.get("userId"),
+            "groupId": payload.get("groupId"),
+        }
+        return self.client.update_request(
+            url_path=f"users/{user_id}/groups/{group_id}", data=data
+        )
```
Comment (auslin-aot) [inline forms-flow-api/src/formsflow_api/services/factory/keycloak_group_service.py:157]: Whether clientId, groupId in the payload & path param is the same?

File: forms-flow-api/src/formsflow_api/services/factory/keycloak_client_service.py [hunk 2]
```diff
@@ -69,3 +69,29 @@ def create_group_role(self, data: Dict):
...
+            "name": payload.get("name"),
+        }
+        return self.client.create_request(
+            url_path=f"users/{user_id}/role-mappings/clients/{client_id}", data=[data]
+        )
+
+    def remove_user_from_group_role(
+        self, user_id: str, group_id: str, payload: Dict = None
+    ):
+        """Remove user to role."""
```
Comment (auslin-aot) [inline forms-flow-api/src/formsflow_api/services/factory/keycloak_client_service.py:88]: please check the typo error in the comment.

File: forms-flow-api/src/formsflow_api/services/factory/keycloak_group_service.py (hunk 1 above)
Comment (sreehari-aot) [reply forms-flow-api/src/formsflow_api/services/factory/keycloak_group_service.py:157]: yes

**Comment by abhilash-aot** [comment]
Resolve conflicts

**Pull request closed without merging**